rust/smoketest/src/tests/test1.rs

```diff
@@ -0,0 +1,18 @@
+/// Check the server sending the HEAD request
+pub async fn check_server(cli: &reqwest::Client, server: &String) -> bool {
```

Comment:
is there a reason why you return true or false over let it panic and use `assert_*` macros?